packages/x-utils/src/get-safe-property-chain.ts

```diff
@@ -19,11 +24,11 @@ export function getSafePropertyChain<T = any>(obj: any, propertyChain: string):
  * @returns The value of the chain of properties, or the default return if any of the intermediate
  * values is not defined.
  */
-export function getSafePropertyChain<T = any>(
-  obj: any,
-  propertyChain: string,
-  defaultReturn?: T
-): T;
+export function getSafePropertyChain<SomeObject, Path extends ExtractPath<SomeObject>>(
+  obj: SomeObject,
+  propertyChain: Path | '',
+  defaultReturn?: ExtractType<SomeObject, Path>
+): ExtractType<SomeObject, Path>;
```

Comment:
Should this overload just handle the empty string version? Why add both?
```suggestion
export function getSafePropertyChain<SomeObject>(
  obj: SomeObject,
  propertyChain: '',
  defaultReturn?: SomeObject
): SomeObject;
```

Reply:
👍🏻